macros/draggableProof.pl

```diff
@@ -1,284 +1,455 @@
-  # been saved, use the complement of the right column.
-  $previous = $main::inputs_ref->{$self->{srcAns}} || "$self->{id}-".join(",$self->{id}-",@diff);
-  $previous =~ s/$self->{id}-//g; $self->{previousSource} = [split(/,/,$previous)];
-}
+$draggable = DraggableProof(
+$statements,
+$extra,
+NumBuckets => 2, # either 1 or 2.
+SourceLabel => "Axioms", # label of first bucket if NumBuckets = 2.
+#
+TargetLabel => "<strong>Reasoning</strong>",
+# label of second bucket if NumBuckets = 2,
+# of the only bucket if NumBuckets = 1.
+################################################################
+# Levenshtein => 1,
+# If equal to 1, scoring is determined by the Levenshtein edit distance between student answer and correct answer.
+################################################################
+# DamerauLevenshtein => 1,
+# If equal to 1, scoring is determined by the Damerau-Levenshtein distance between student answer and correct answer.
+# A pair of transposed adjacent statements is counted as two mistakes under Levenshtein scoring,
+# but as one mistake under Damerau-Levenshtein scoring.
+################################################################
+InferenceMatrix => [
+[0, 0, 1],
+[0, 0, 1],
+[0, 0, 0]
+],
+# (i, j)-entry is nonzero <=> statement i implies statement j.
+# The score of each corresponding inference is weighted according to the value of the matrix entry.
+################################################################
+IrrelevancePenalty => 1 # This option is processed only if the InferenceMatrix option is set.
+# Penalty for each extraneous statement in the student answer is <IrrelevancePenalty>
+# divided by the total number of inference points (i.e. sum of all entries in the InferenceMatrix).
+# Default value = 1.
+);
 
-sub Print {
-  my $self = shift;
+Context()->texStrings; # must be placed AFTER defining all DraggableSubsets.
 
-  if ($main::displayMode ne "TeX") { # HTML mode
+BEGIN_TEXT
 
-    return join("\n",
-       '<div style="min-width:750px;">',
-          $self->Source,
-          $self->Target,
-       '<br clear="all" />',
-      '</div>',
-    );
+Show that Socrates is mortal by dragging the relevant $BBOLD Axioms $EBOLD
+into the $BBOLD Reasoning $EBOLD box in an appropriate order.
 
-  } else { # TeX mode
+$PAR
 
-    return join("\n",
-          $self->Source,
-          $self->Target,
-    );
+\{ $draggable->Print \}
 
-  }
+END_TEXT
+Context()->normalStrings;
+
+ANS($draggable->cmp);
+
+ENDDOCUMENT();
+
+=cut
+
+################################################################
+
+loadMacros(
+"PGchoicemacros.pl",
+"MathObjects.pl",
+);
 
+sub _draggableProof_init {
+	ADD_CSS_FILE("https://cdnjs.cloudflare.com/ajax/libs/nestable2/1.6.0/jquery.nestable.min.css", 1);
+	ADD_JS_FILE("https://cdnjs.cloudflare.com/ajax/libs/nestable2/1.6.0/jquery.nestable.min.js", 1);
+	ADD_CSS_FILE("js/apps/DragNDrop/dragndrop.css", 0);
+	ADD_JS_FILE("js/apps/DragNDrop/dragndrop.js", 0, { defer => undef });
```

Comment:
I just noticed that you defer loading of the dragndrop.js file, but you do not defer loading of the jquery.nestable.js file.  You should also defer loading for that.